interfaces/repo.go

```diff
@@ -322,68 +322,100 @@ func (r *Repository) Connect(plugSnapName, plugName, slotSnapName, slotName stri
 //   from all the slots found therein. It is not an error if there are no
 //   such plugs but it is still an error if the snap does not exist or has no
 //   slots at all.
-func (r *Repository) Disconnect(plugSnapName, plugName, slotSnapName, slotName string) error {
+//
+// Disconnect returns the list of severed connections, the list of affected snaps
+// and an error if one occurred.
+func (r *Repository) Disconnect(plugSnapName, plugName, slotSnapName, slotName string) ([]ConnRef, []*snap.Info, error) {
```

Comment:
It's also not clear to me if it makes sense to be returning those snap.Infos.

@zyga didn't we have bugs because the snap.Info here is cached and potentially out-of-date?  isn't it dangerous to be exposing it?  Do we even need to expose it?

We also need to update documentation to reflect those results.